cmd/icingadb/main.go

```diff
@@ -166,10 +166,15 @@ func run() int {
 						synctx, cancelSynctx := context.WithCancel(ha.Environment().NewContext(hactx))
 						g, synctx := errgroup.WithContext(synctx)
 						// WaitGroups for initial synchronization.
-						// Runtime updates must wait for initial synchronization to complete.
+						// Runtime updates and history pipelines must wait for the initial synchronization to
+						// complete by draining the `initConfigSyncDone` channel.
 						configInitSync := sync.WaitGroup{}
 						stateInitSync := &sync.WaitGroup{}
 
+						// A buffered channel used to notify both the runtime updates and history pipelines workers
+						// about the successful initial config sync completion including the SLA lifecycles.
+						initConfigSyncDone := make(chan struct{}, 1)
```

Comment:
Why aren't you using a `WaitGroup` here as well? As far as I can see, this channel is just used as a semaphore.

Reply:
> Why aren't you using a `WaitGroup` here as well?

Because there are no worker groups I need to wait for! There is only a single worker that writes into that channel, so I don't need a `WaitGroup` for it.

> As far as I can see, this channel is just used as a semaphore.

Semaphore? I don't see how this relates to semaphores in any way, I only have a single worker writing just a **single** time in it to unlock the waiters.

Reply:
Pardon my French, but semaphore as in a lock for a resource.

Back to your diff: Why are you creating a buffered channel and why are you writing something into it before closing? Unless I am missing something, you can just use an unbuffered channel, wait for it in the consumers and close it in the producer. For example <https://go.dev/play/p/78zLpPxpMsW>.